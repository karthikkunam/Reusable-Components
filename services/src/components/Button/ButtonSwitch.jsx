import React, { useState } from 'react';
import PropTypes from 'prop-types';

const ButtonSwitch = props => {
  const [value, setValue] = useState(props.defaultView ? props.defaultView : props.buttonOptions[0].value);
  const { buttonOptions, onClick, className, size } = props;

  const onButtonSwitch = value => {
    setValue(value);
    onClick && onClick(value);
  };
  return (
    <div className={`component-example mb-3 ${className}`}>
      <div
        className={`btn-group btn-group-toggle ${size === 'lg' ? 'btn-group-lg' : 'btn-group-sm'}`}
        data-toggle="buttons">
        {buttonOptions &&
          buttonOptions.length > 0 &&
          buttonOptions.map((option, index) => {
            return (
              <label key={index} className={`btn btn-secondary ${option.value === value ? 'active' : ''}`}>
                <input
                  type="radio"
                  name="options"
                  id={option.id}
                  autoComplete="off"
                  checked={option.value === value}
                  onClick={() => {
                    onButtonSwitch(option.value);
                  }}
                />
                {option.label}
              </label>
            );
          })}
      </div>
    </div>
  );
};

ButtonSwitch.displayName = 'ButtonSwitch';
ButtonSwitch.defaultProps = {
  onClick: () => {},
  className: '',
};

ButtonSwitch.propTypes = {
  buttonOptions: PropTypes.arrayOf({
    id: PropTypes.string, 
    label: PropTypes.string, 
    value: PropTypes.string,
  }).isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  defaultView: PropTypes.string,
  size: PropTypes.oneOf(['lg']),
};

export default ButtonSwitch;
