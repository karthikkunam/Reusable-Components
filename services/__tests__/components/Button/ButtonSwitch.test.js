import React from 'react';
import { shallow } from 'enzyme';
import ButtonSwitch from '../../../src/components/Button/ButtonSwitch.jsx';

const props = {
  buttonOptions: [
    { id: '1', label: 'Cost ($)', value: 'Cost' },
    { id: '2', label: 'Retail ($)', value: 'Retail' },
  ],
  onClick: jest.fn(() => {}),
  className: '',
};
describe('ButtonSwitch Component', () => {
  it('renders correctly', () => {
    const component = shallow(<ButtonSwitch {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders secondaryView as defaultView', () => {
    const props1={
      defaultView: 'Retail',
    };
    const component = shallow(<ButtonSwitch {...props} {...props1} />);
    expect(component).toMatchSnapshot();
  });
  it('renders size prop', () => {
    const props1={
      size: 'lg',
    };
    const component = shallow(<ButtonSwitch {...props} {...props1} />);
    expect(component).toMatchSnapshot();
  });
  it('should swich buttons', () => {
    const component = shallow(<ButtonSwitch {...props} />);
    component
      .find('input')
      .at(0)
      .simulate('click', {
        onButtonSwitch: () => {}
      });
    expect(component).toMatchSnapshot();
  });
});
