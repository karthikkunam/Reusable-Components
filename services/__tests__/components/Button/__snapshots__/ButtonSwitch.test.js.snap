// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSwitch Component renders correctly 1`] = `
<div
  className="component-example mb-3 "
>
  <div
    className="btn-group btn-group-toggle btn-group-sm"
    data-toggle="buttons"
  >
    <label
      className="btn btn-secondary active"
      key="0"
    >
      <input
        autoComplete="off"
        checked={true}
        id="1"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Cost ($)
    </label>
    <label
      className="btn btn-secondary "
      key="1"
    >
      <input
        autoComplete="off"
        checked={false}
        id="2"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Retail ($)
    </label>
  </div>
</div>
`;

exports[`ButtonSwitch Component renders secondaryView as defaultView 1`] = `
<div
  className="component-example mb-3 "
>
  <div
    className="btn-group btn-group-toggle btn-group-sm"
    data-toggle="buttons"
  >
    <label
      className="btn btn-secondary "
      key="0"
    >
      <input
        autoComplete="off"
        checked={false}
        id="1"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Cost ($)
    </label>
    <label
      className="btn btn-secondary active"
      key="1"
    >
      <input
        autoComplete="off"
        checked={true}
        id="2"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Retail ($)
    </label>
  </div>
</div>
`;

exports[`ButtonSwitch Component renders size prop 1`] = `
<div
  className="component-example mb-3 "
>
  <div
    className="btn-group btn-group-toggle btn-group-lg"
    data-toggle="buttons"
  >
    <label
      className="btn btn-secondary active"
      key="0"
    >
      <input
        autoComplete="off"
        checked={true}
        id="1"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Cost ($)
    </label>
    <label
      className="btn btn-secondary "
      key="1"
    >
      <input
        autoComplete="off"
        checked={false}
        id="2"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Retail ($)
    </label>
  </div>
</div>
`;

exports[`ButtonSwitch Component should swich buttons 1`] = `
<div
  className="component-example mb-3 "
>
  <div
    className="btn-group btn-group-toggle btn-group-sm"
    data-toggle="buttons"
  >
    <label
      className="btn btn-secondary active"
      key="0"
    >
      <input
        autoComplete="off"
        checked={true}
        id="1"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Cost ($)
    </label>
    <label
      className="btn btn-secondary "
      key="1"
    >
      <input
        autoComplete="off"
        checked={false}
        id="2"
        name="options"
        onClick={[Function]}
        type="radio"
      />
      Retail ($)
    </label>
  </div>
</div>
`;
